src/components/verification-grid/verification-grid.ts

```diff
@@ -0,0 +1,1364 @@
+    this.undecidedTiles = [];
+    if (hasSubSelection) {
+      this.undecidedTiles = gridTiles
+        .filter((tile: VerificationGridTileComponent) => !subSelection.includes(tile))
+        .map((tile: VerificationGridTileComponent) => tile.model);
+    }
+
+    if (this.isViewingHistory()) {
+      // when viewing history, we don't want to add the decision to the history
+      // we want to update the decision that was made
+      const decisionsToUpdate = this.decisions.filter((decision: DecisionWrapper) =>
+        selectedItems.some((item: DecisionWrapper) => item.url === decision.url),
+      );
+
+      decisionsToUpdate.forEach((historicalDecision: DecisionWrapper) => {
+        historicalDecision.decisions = classifications;
+      });
+
+      // we have updated the decision about a tiles while viewing history
+      // therefore, there will be an already existing outline around the
+      // grid tile that we need to update
+      this.createSelectionHighlight(selectedTiles, decisionColor);
+
+      this.removeDecisionButtonHighlight();
+      this.showDecisionButtonHighlight([event.detail.target]);
+
+      // by returning early, we prevent the decision from being added to the
+      // history and do not page forward
+      return;
+    }
+
+    // if we are in the normal paging scenario, we want to add the decision to
+    // the decision history as a new decision
+    this.decisions.push(...decisions);
+    this.dispatchEvent(new CustomEvent(VerificationGridComponent.decisionMadeEventName, { detail: decisions }));
+
+    if (this.autoPage) {
+      this.showDecisionButtonHighlight([event.target as any]);
+      this.createSelectionHighlight(selectedTiles, decisionColor);
+
+      // we wait for 300ms so that the user has time to see the decision that
+      // they have made in the form of a decision highlight around the selected
+      // grid tiles and the chosen decision button
+      await sleep(300);
+      this.removeDecisionHighlight(selectedTiles);
+
+      this.nextPage(decisions.length);
+    }
+  }
+
+  private setDecisionDisabled(disabled: boolean): void {
+    const decisionElements = this.decisionElements ?? [];
+    for (const decisionElement of decisionElements) {
+      decisionElement.disabled = disabled;
+    }
+  }
+
+  //#endregion
+
+  //#region DecisionHighlights
+
+  private decisionColor(verification: DecisionWrapper): string {
+    // TODO: this is really slow, we should probably optimize it
+    const decisionElements = this.decisionElements ?? [];
+    const decisionButton = decisionElements.find((element: DecisionComponent) => {
+      const verificationDecisions = verification.decisions;
+      const elementModels = element.classificationModels();
+      return JSON.stringify(verificationDecisions) === JSON.stringify(elementModels);
+    });
+
+    if (!decisionButton) {
+      throw new Error("Could not find color to match decision");
+    }
+
+    return decisionButton.color;
+  }
```

Comment:
okay i think this can be structured better.

if the decisions for a particular session have an index, we can tag models with the decision that was made (say with a symbol), then when a tile is rendered, it can apply a class like `.decision-1` whenever it's state is set to show decision.

So no more searching through decision elements to get colors.